src/sentry/incidents/logic.py

```diff
@@ -803,6 +822,15 @@ def delete_alert_rule(alert_rule, user=None):
         # Change the incident status asynchronously, which could take awhile with many incidents due to snapshot creations.
         tasks.auto_resolve_snapshot_incidents.apply_async(kwargs={"alert_rule_id": alert_rule.id})
 
+    if user:
+        create_audit_entry_from_user(
+            user,
+            organization_id=alert_rule.organization_id,
+            target_object=alert_rule.id,
```

Comment:
When we call `alert_rule.delete()` the `id` will be set to `None`. We should cache it earlier, before we run the deletion. It might actually make sense to just include this audit creation inside of the transaction above